### PR TITLE
Feature/client testing

### PR DESCRIPTION
**Pull request by AltusBarry**
Tests for the client code.

Small refactor of client code and updates to support this where needed.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -53,11 +53,17 @@ Installation
     These can also be specificied per field by passing the ``public_key`` or
     ``private_key`` parameters to ``ReCaptchaField`` - see field usage below.
 
-#. If you require a proxy, add a ``RECAPTCHA_PROXY`` setting, for example:
+#. If you require a proxy, add a ``RECAPTCHA_PROXY`` setting dictionary:
```
Comment (cblignaut) [inline README.rst]: setting (dictionary of proxies), for example

File: captcha/__init__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from django.conf import settings
```
Comment (cblignaut) [inline captcha/__init__.py]: I would rather create a default config in apps.py for this

Comment (AltusBarry) [reply captcha/__init__.py]: I'll meet you halfway and setup default config in here as well. I want these values checked on package initialisation.

File: README.rst (hunk 1 above)
Comment (cobusc) [reply README.rst]: @cblignaut minor correction: that is a set, not a dictionary.

Comment (cobusc) [reply README.rst]: @AltusBarry how does the code know which proxy to use? (I'll read a bit further. MAybe I get the answer myself.)

File: captcha/client.py [hunk 3]
```diff
@@ -3,104 +3,69 @@
...
+            "User-agent": "reCAPTCHA Django"
+        }
+    )
 
-        return opener.open(*args, **kwargs)
-    else:
-        return urlopen(*args, **kwargs)
+    # Add proxy values to opener if needed.
+    opener_args = []
+    proxy_settings = getattr(settings, "RECAPTCHA_PROXY", {})
```
Comment (cobusc) [inline captcha/client.py]: How about renaming `proxy_settings` to `proxies`, which is more accurate IMHO?

File: captcha/client.py [hunk 4]
```diff
@@ -3,104 +3,69 @@
...
-    verify_url = 'https://%s/recaptcha/api/siteverify' % VERIFY_SERVER
-    req = Request(
-        url=verify_url,
-        data=params,
-        headers={
-            'Content-type': 'application/x-www-form-urlencoded',
-            'User-agent': 'reCAPTCHA Python'
-        }
+    if not PY2:
+        params = params.encode("utf-8")
```
Comment (cobusc) [inline captcha/client.py]: Whitespace after this line, please.

File: README.rst (hunk 1 above)
Comment (cobusc) [reply README.rst]: @cblignaut Apologies. I see it _should_ be a dictionary, but is incorrectly defined as a set below, hence my comment.

File: captcha/__init__.py (hunk 2 above)
Comment (AltusBarry) [reply captcha/__init__.py]: I personally don't like proxy variables for settings. Code base usually ends up in a position where there is a need to check against django settings or the default conf depending on the logic you want/need.

Just stick to settings and if needed, set the defaults on a per module level using constants to store the defaults.

**Pull request closed without merging**